### PR TITLE
Add Emscripten Math API for direct access to JavaScript Math object

### PR DESCRIPTION
**Pull request by juj**
This help codebases explicitly choose when to opt in for the JS `Math` API, to get small size (and perhaps higher precision?)

### CODE REVIEW

**Comment by juj** [comment]
In my case, I want ot call `Math.atan2` instead of `atan2`, because the libc version brings in monstrous 5KB of `rem_pio` stuff; but keep other functions intact, which are not growing the size of my binary as much.

**Comment by sbc100** [comment]
> In my case, I want ot call `Math.atan2` instead of `atan2`, because the libc version brings in monstrous 5KB of `rem_pio` stuff; but keep other functions intact, which are not growing the size of my binary as much.

And would the `JS_MATH` option be OK for you in this case, or it important the the other function do *not* from JS?

Maybe its worth tuning the behaviour of JS_MATH to have this behavior if there are clear separation  between function it makes sense get from JS and function that it makes sense to keep native?    

Another way of putting it, if JS_MATH isn't good enough, maybe lets make it more useful?

**Comment by juj** [comment]
I don't like to use JS_MATH for this. Having the flexibility to choose per call seems valuable to me, it is more future proof as well, in case the landscape changes on us - e.g. one day we conclude that `atan` via math.h would be worth the code size, whereas some other functions aren't. 

Further, being able to control individual functions lets one profile what the effect on size or performance is per function or per call site, that feels like it would provide "most power" to user side for all use cases, rather than enforcing Emscripten to make a fixed rule of "all or nothing", or "a balance that was tuned on one project that Emscripten authors happened to have visibility on at the time the JS_MATH feature was crafted".

Having an option to route all libc math to JS-side can be a good feature for quick wholesale testing, and when not having much control over the whole codebase that is being compiled, so there is benefit in having that. But I do see much benefit in having direct intrinsics like functions as well.

File: system/include/emscripten/math.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2012 The Emscripten Authors.  All rights reserved.
```
Comment (kripken) [inline system/include/emscripten/math.h:2]: `2020`

**Pull request merged**